sshd/issue_invite.sh

```diff
@@ -67,18 +68,8 @@ cat $TMP/id_rsa.pub >> $HOMEDIR/.ssh/authorized_keys
 
 # Output the actual invite code.
 PUBLIC_IP=`cat /hostname`
-export CLOUD_INSTANCE_DETAILS=$(cat << EOF
-{
-  "host":"$PUBLIC_IP",
-  "user":"$USERNAME",
-  "key":"$ENCODED_KEY"
-}
-EOF
-)
+export CLOUD_INSTANCE_DETAILS="\{\"host\":\"$PUBLIC_IP\",\"user\":\"$USERNAME\",\"key\":\"$ENCODED_KEY\"\}"
 
-echo|base64 -w 0 << EOF
-{
-  "networkName":"Cloud",
-  "networkData":"`echo -n $CLOUD_INSTANCE_DETAILS|sed s/'"'/'\\\\"'/g`"
-}
-EOF
+CLOUD_INSTANCE_DETAILS=`nodejs -p -e "require('jsurl').stringify('$CLOUD_INSTANCE_DETAILS');"`
+
+echo "v=2&networkName=Cloud&networkData=$CLOUD_INSTANCE_DETAILS&userName=$USERNAME"
```

Comment:
For other social networks, we need a userName because we display a message like "Bob has invited you to uProxy".  For cloud we have some special case code that just sets the userName to "a virtual machine" (https://github.com/uProxy/uproxy/blob/master/src/generic_ui/scripts/ui.ts#L523).  I think otherwise userName won't be used by cloud invites so it should be fine leaving it out